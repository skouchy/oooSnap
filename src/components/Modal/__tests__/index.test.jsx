/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Modal from "..";

const currentPhoto = {
    name: "Park Bench",
    category: "landscape",
    description: "Lorem ipsum dolor",
    index: 1,  
};

const mockToggleModal = jest.fn();

describe("Modal component", () => {
  it("renders", () => {
    render(<Modal onClose={mockToggleModal} currentPhoto={currentPhoto}/>);
  });

  it("matches snapshot DOM node structure", () => {
    const { asFragment } = render(<Modal onClose={mockToggleModal} currentPhoto={currentPhoto} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Click Event", () => {
  it("calls onClose handler", () => {
    // Arrange: Render Modal
    const { getByText } = render(<Modal onClose={mockToggleModal} currentPhoto={currentPhoto} />);
    
    // Act: Simulate click event
    fireEvent.click(getByText('Close this Modal'));
    
    // Assert: Expected matcher
    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
});

afterEach(cleanup);
