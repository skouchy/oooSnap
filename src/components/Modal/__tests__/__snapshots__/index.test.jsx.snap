// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park Bench
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        Lorem ipsum dolor
      </p>
      <button
        data-testid="close"
        type="button"
      >
        Close this Modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
